subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/processors/MaxNParallelTestClassProcessor.java

```diff
@@ -74,9 +75,11 @@ public void processTestClass(TestClassRunInfo testClass) {
             processors.add(processor);
             processor.startProcessing(resultProcessor);
         } else {
-            processor = processors.get(pos);
-            pos = (pos + 1) % processors.size();
+            int shift = (testCount /maxProcessors) % maxProcessors;
```

Comment:
Did you mean space between `/` and `maxProcessors`?

Reply:
well spotted :)